frontend/components/Course/Analytics/Heatmaps/Heatmap.tsx

```diff
@@ -0,0 +1,74 @@
+import dynamic from "next/dynamic";
+import React from "react";
+import { HeatmapSeries } from "../../../../types";
+
+interface HeatmapProps {
+    series: HeatmapSeries[];
+    chartTitle: string;
+}
+
+const Chart = dynamic(() => import("react-apexcharts"), { ssr: false });
```

Comment:
Not that familiar with Next js dynamic importing, but it seems like react apex charts doesn't work on the server (get "window is not defined" at apexcharts/dist/apexcharts.common.js if importing it normally). I think it might be an import error since not even putting a window !== undefined check around the chart component doesn't work. I saw "when the module includes a library that only works in the browser" was one of the use cases for dynamic imports

Let me know if there's something better though!

Reply:
Nope, this sounds like the perfect use case for dynamic imports - just leave a small comment in code for future purposes.